setup.py

```diff
@@ -1,9 +1,32 @@
-from os.path import basename
-from setuptools import setup, find_packages
+import sys
+import importlib
 from glob import glob
+from os.path import basename
+from setuptools import setup, find_packages, Extension, _install_setup_requires
+
+
+try:
+    from distutils.config import ConfigParser
+except ImportError:
+    from configparser import ConfigParser
+
+conf = ConfigParser()
+conf.read(['setup.cfg'])
 
+# Get some config values
+metadata = dict(conf.items('metadata'))
+PACKAGENAME = metadata['name']
+
+options = dict(conf.items('options'))
+SETUP_REQUIRES = [
+    s for s in map(str.strip, options['setup_requires'].splitlines()) if s
+]
+INSTALL_REQUIRES = [
+    s for s in map(str.strip, options['install_requires'].splitlines()) if s
+]
```

Comment:
This is all handled from pyproject.toml build backend, and already currently works. I.e. you don't need to parse setup.cfg.  This already happens because

```
build-backend = "setuptools.build_meta"
```